conwayste/src/ui/helpe.rs

```diff
@@ -0,0 +1,151 @@
+/*  Copyright 2019 the Conwayste Developers.
+ *
+ *  This file is part of conwayste.
+ *
+ *  conwayste is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  conwayste is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with conwayste.  If not, see
+ *  <http://www.gnu.org/licenses/>. */
+
+use std::rc::Rc;
+
+use ggez::graphics::{self, Font, Rect, Text, TextFragment, DrawParam, Color};
+use ggez::nalgebra::Point2;
+use ggez::{Context, GameResult};
+
+use crate::constants::DEFAULT_UI_FONT_SCALE;
+
+/// Helper function to draw text onto the screen.
+/// Given the string `str`, it will be drawn at the point coordinates specified by `coords`.
+/// An offset can be specified by an optional `adjustment` point.
+///
+/// # Return value
+///
+/// On success, an `Ok((text_width, text_height))` tuple is returned, indicating the width
+/// and height of the text in pixels.
+pub fn draw_text(ctx: &mut Context, font: Rc<Font>, color: Color, text: String, coords: &Point2<f32>) -> GameResult<(f32, f32)> {
```

Comment:
I'd suggest some new arguments
* bounds: Rect
* an enum to indicate how to handle the text being too big for the width/height (wrap, cut off, etc.)

Reply:
See comment regarding label API.